src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,922 @@
+        else:
+            return consts.Diagnostic_Check_Incomplete, storage_space_available
+
+        # If both the check passed then we will return Diagnoser checks Passed
+        if(dns_check == consts.Diagnostic_Check_Passed and outbound_connectivity_check == consts.Diagnostic_Check_Passed):
+            return consts.Diagnostic_Check_Passed, storage_space_available
+        # If any of the check remain Incomplete than we will return Incomplete
+        elif(dns_check == consts.Diagnostic_Check_Incomplete or outbound_connectivity_check == consts.Diagnostic_Check_Incomplete):
+            return consts.Diagnostic_Check_Incomplete, storage_space_available
+        else:
+            return consts.Diagnostic_Check_Failed, storage_space_available
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+        logger.warning("An exception has occured while trying to perform diagnoser container check on the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Diagnoser_Container_Check_Failed_Fault_Type, summary="Error occured while performing the diagnoser container checks")
+        diagnoser_output.append("An exception has occured while trying to perform diagnoser container check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return consts.Diagnostic_Check_Incomplete, storage_space_available
+
+
+def executing_diagnoser_job(corev1_api_instance, batchv1_api_instance, filepath_with_timestamp, storage_space_available, absolute_path, helm_client_location, kubectl_client_location, release_namespace, probable_pod_security_policy_presence):
+
+    global diagnoser_output
+    job_name = "azure-arc-diagnoser-job"
+    # CMD command to get helm values in azure arc and converting it to json format
+    command = [helm_client_location, "get", "values", "azure-arc", "--namespace", release_namespace, "-o", "json"]
+    # Using Popen to execute the helm get values command and fetching the output
+    response_helm_values_get = Popen(command, stdout=PIPE, stderr=PIPE)
+    output_helm_values_get, error_helm_get_values = response_helm_values_get.communicate()
+    if response_helm_values_get.returncode != 0:
+        if ('forbidden' in error_helm_get_values.decode("ascii") or 'timed out waiting for the condition' in error_helm_get_values.decode("ascii")):
+            telemetry.set_exception(exception=error_helm_get_values.decode("ascii"), fault_type=consts.Get_Helm_Values_Failed,
+                                    summary='Error while doing helm get values azure-arc')
+    helm_values_json = json.loads(output_helm_values_get)
+    # Retrieving the proxy values if they are present
+    try:
+        is_proxy_enabled = helm_values_json["global"]["isProxyEnabled"]
+    except Exception as e:
+        if 'isProxyEnabled' in str(e):
```

Comment:
Add a comment for this that this exception will come whenever the field is not present and that is why we're doing this check before setting it to false